aggregator/internal/pkg/aggregator.go

```diff
@@ -282,7 +282,18 @@ func (agg *Aggregator) handleBlsAggServiceResponse(blsAggServiceResp blsagg.BlsA
 			agg.logger.Info("Aggregator successfully responded to task",
 				"taskIndex", blsAggServiceResp.TaskIndex,
 				"batchIdentifierHash", "0x"+hex.EncodeToString(batchIdentifierHash[:]))
-				
+
+			// If Aggregator successfully responds to task we acquire the task mutex and
+			// remove task information from the aggregator maps to prevent a memory leak.
+			agg.taskMutex.Lock()
+			agg.AggregatorConfig.BaseConfig.Logger.Info("- Locked Resources: Removing Task Info from Aggregator")
```

Comment:
The logs should probably be outside the critical section.

Reply:
Addessed, thank you @Oppen